src/features/ScreenLock/ScreenLockProvider.tsx

```diff
@@ -261,12 +266,24 @@ class ScreenLockProvider extends Component<RouteComponentProps & IDataContext &
 const mapStateToProps = (state: AppState) => ({
   exportState: exportState(state),
   isEncrypted: isEncrypted(state),
-  getEncryptedData: getEncryptedData(state)
+  getEncryptedData: getEncryptedData(state),
+  password: getPassword(state),
+  inactivityTimer: getInactivityTimer(state)
 });
 const mapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
-  bindActionCreators({ importState, clearEncryptedData, encrypt, decrypt }, dispatch);
+  bindActionCreators(
+    {
+      importState,
+      clearEncryptedData,
+      encrypt,
+      decrypt,
+      setPassword,
+      appReset: appResetAction
+    },
+    dispatch
+  );
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 type Props = ConnectedProps<typeof connector>;
 
-export default pipe(withRouter, withContext(DataContext), connector)(ScreenLockProvider);
+export default withRouter(connector(ScreenLockProvider));
```

Comment:
do we want to keep `withRouter` rather than using hooks ?

Reply:
For the moment the component is still a class and a refactor to a function component is out of scope